lib/google/ads/google_ads/google_ads_client.rb

```diff
@@ -117,7 +137,7 @@ def endpoint
 
         def target
           default_target = "googleads.googleapis.com:443"
-          target = ENV.fetch('GOOGLEADS_SERVICE_PATH', default_target)
+          target = ENV.fetch('GOOGLE_ADS_ENDPOINT', default_target)
```

Comment:
This seems to always be loaded even when `configure_using_environment_variables` is never called. You may need to redesign this to use a new configuration setting instead.

Reply:
Changed to a new design to read from config setting.

Reply:
This one should *always* load if specified, regardless of config setting. It's only for internal development, so it won't cause any issues for end users.

Reply:
Good point, @mcloonan . A few things:

1. It makes better sense to use `GOOGLE_ADS_CONFIGURATION_FILE_PATH` to switch between configurations instead of loading the endpoint all the time.
1. @wfansh and I were discussing the possibility to add the "load from environment variables" call to all code examples for your use case but also our functional tests. We could make this call optional with an additional argument in all code examples. @AnashOommen WDYT?

Reply:
1. We specifically chose not to allow setting the endpoint via the config so that users don't think it's something they should do. This is something that only maintainers of the library can ever use, so it doesn't need to be "user-friendly" in the normal sense.
2. I'm strongly opposed to modifying every example for any reason at this point.

Reply:
After some discussion, this looks good as is. Allow configuring the endpoint in the config and via environment variable only when explicitly loading the env config.

I'm still opposed to modifying every example, though :)

Reply:
Seems like we have a consensus :) I'll resolve this thread and keep the current implementation.